docs/community-toolkit/hosting-deno.md

```diff
@@ -0,0 +1,93 @@
+---
+title: .NET Aspire Community Toolkit Deno hosting integration
+description: Learn about the .NET Aspire Community Toolkit Deno hosting extensions package which provides functionality to run Deno applications and tasks.
+ms.date: 10/25/2024
+---
+
+# .NET Aspire Community Toolkit Deno hosting integration
+
+[!INCLUDE [includes-hosting](../includes/includes-hosting.md)]
+
+[!INCLUDE [banner](includes/banner.md)]
+
+In this article, you learn about the .NET Aspire Community Toolkit Deno package. The extensions package brings the following features:
+
+- Running [Deno](https://deno.com/) applications
+- Running Node.js applications via Deno tasks
+- Ensuring that the packages are installed before running the application via Deno installer
+
+## Hosting integration
+
+To get started with the .NET Aspire Community Toolkit Deno extensions, install the [📦 CommunityToolkit.Aspire.Hosting.Deno](https://nuget.org/packages/CommunityToolkit.Aspire.Hosting.Deno) NuGet package in the AppHost project.
+
+### [.NET CLI](#tab/dotnet-cli)
+
+```dotnetcli
+dotnet add package CommunityToolkit.Aspire.Hosting.Deno
+```
+
+### [PackageReference](#tab/package-reference)
+
+```xml
+<PackageReference Include="CommunityToolkit.Aspire.Hosting.Deno"
+                  Version="*" />
+```
+
+---
+
+For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).
+
+## Example usage
+
+The following sections detail various usages, from running Vite applications to using specific package managers.
+
+### Run Deno apps
+
+This integration extension adds support for running a Deno application defined in a script. Since [Deno is secure by default](https://docs.deno.com/runtime/fundamentals/security), permission flags must be specified in `permissionFlags` argument of `AddDenoApp`.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddDenoApp("oak-demo", "main.ts", permissionFlags: ["-E", "--allow-net"])
```

Comment:
I don't know Deno very well, so maybe this is idiomatic as is, but as a newcomer I'd find `--allow-env` more helpful than `-E`.

They're equivalent according to the docs: https://docs.deno.com/runtime/fundamentals/security/#environment-variables

Reply:
What do you think of me showing both the long form and short form flags?

Reply:
I saw 0433682 - but I'm not a fan of both examples in the code like that. I'd instead include text that says that the preceding code demonstrates passing the `permissionFlags` with their fully qualified switches, and that you could just as easily use the short-hand aliases and then list them there.